src/inspection/inspectionSelectDates.tsx

```diff
@@ -0,0 +1,140 @@
+import { InspectionDate, InspectionInput, InspectionType } from '../schema-types'
+import Dropdown from '../common/input/Dropdown'
+import { getDateObject, readableDateRange } from '../util/formatDate'
+import { useLazyQueryData } from '../util/useLazyQueryData'
+import { allInspectionDatesQuery } from './inspectionDate/inspectionDateQuery'
+import { LoadingDisplay } from '../common/components/Loading'
+import { text } from '../util/translate'
+
+const InspectionSelectDatesView = styled.div`
+  margin: 1rem 0;
+  width: 50%;
+`
+
+interface DateOption {
+  label: string
+  value: {
+    startDate: Date
+    endDate: Date
+  }
+}
+
+export type PropTypes = {
+  inspectionType: InspectionType
+  inspectionInput: InspectionInput
+  onChange: (startDate: Date, endDate: Date) => void
+}
+
+const InspectionSelectDates = observer(
+  ({ inspectionType, inspectionInput, onChange }: PropTypes) => {
+    let [
+      queryInspectionDates,
+      { data: inspectionDatesQueryResult, loading: areInspectionDatesLoading },
+    ] = useLazyQueryData<InspectionDate[]>(allInspectionDatesQuery)
+
+    let _queryPostinspectionDateOptions = async () => {
+      if (!areInspectionDatesLoading && !inspectionDatesQueryResult) {
+        await queryInspectionDates()
+      }
+    }
+
+    let dateOptions: DateOption[] = []
+    if (inspectionType === InspectionType.Pre) {
+      dateOptions = _getPreInspectionDateOptions()
+    } else {
+      if (inspectionDatesQueryResult) {
+        dateOptions = _getPostInspectionDateOptions(inspectionDatesQueryResult)
+      } else {
+        _queryPostinspectionDateOptions()
+      }
+    }
+    dateOptions.sort((a: DateOption, b: DateOption) => {
+      return a.value.startDate.getTime() < b.value.startDate.getTime() ? -1 : 1
+    })
+
+    let onSelectDates = (dateOption: DateOption) => {
+      onChange(dateOption.value.startDate, dateOption.value.endDate)
+    }
+    let selectedItem: DateOption | null =
+      inspectionInput.inspectionStartDate && inspectionInput.inspectionEndDate
+        ? {
+            label: readableDateRange({
+              startDate: inspectionInput.inspectionStartDate,
+              endDate: inspectionInput.inspectionEndDate,
+            }),
+            value: {
+              startDate: inspectionInput.inspectionStartDate,
+              endDate: inspectionInput.inspectionEndDate!,
+            },
+          }
+        : null
+    return (
+      <InspectionSelectDatesView>
+        {areInspectionDatesLoading ? (
+          <LoadingDisplay />
+        ) : (
+          <Dropdown
+            label={text('inspection.select_inspection')}
+            items={dateOptions}
+            onSelect={onSelectDates}
+            selectedItem={selectedItem}
+          />
+        )}
+      </InspectionSelectDatesView>
+    )
+  }
+)
+
+const _getPreInspectionDateOptions = (): DateOption[] => {
+  let startDate = new Date()
+  let endDate = new Date(startDate)
+  endDate.setDate(endDate.getDate() + 90)
```

Comment:
Javascript's Date object is horrible and we should use date-fns functions wherever possible. I have no idea what's going on here. Suggestion:

```suggestion
  let endDate = addSeconds(startDate, 90)
```

Substitute "addSeconds" with the function that corresponds to what unit the "90" is supposed to be.